app/models/king.rb

```diff
@@ -1,12 +1,44 @@
 class King < Piece
 
+	def can_castle?(x_destination, y_destination)
+		# Castling allowed if:
+		# king and rook not moved
+		# no pieces between king and rook
+		# king not in check either at start or at destination
+		@game = self.game
+		@has_moved != true && !obstructed?([x_destination, y_destination]) && @game.in_check?(color) == false
```

Comment:
Colin, I'm not sure what you mean... If we change @game.in_check?(color) to
@game.in_check?(current_color), then we'll get an error because at this
point current_color doesn't exist. Only 'color' exists as a piece
attribute. Does that make sense?

On Mon, Aug 3, 2015 at 11:33 AM, Colin Rubbert <notifications@github.com>
wrote:

> In app/models/king.rb
> <https://github.com/StormyKnights/StormyKnights/pull/26#discussion_r36097169>
> :
>
> > @@ -1,12 +1,44 @@
> >  class King < Piece
> >
> > +	def can_castle?(x_destination, y_destination)
> > +		# Castling allowed if:
> > +		# king and rook not moved
> > +		# no pieces between king and rook
> > +		# king not in check either at start or at destination
> > +		@game = self.game
> > +		@has_moved != true && !obstructed?([x_destination, y_destination]) && @game.in_check?(color) == false
>
> If you are changing the in_check?(color) method to
> in_check?(current_color) then this line should be updated because it's
> still using the old method.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StormyKnights/StormyKnights/pull/26/files#r36097169>.
>